src/vm/arm/patchedcode.S

```diff
@@ -0,0 +1,576 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+// ==++==
+//
+
+//
+// ==--==
+#include "unixasmmacros.inc"
+#include "asmconstants.h"
+
+.syntax unified
+.thumb
+
+    SETALIAS JIT_New, _Z7JIT_NewP21CORINFO_CLASS_STRUCT_
+    SETALIAS JIT_Box, _Z7JIT_BoxP21CORINFO_CLASS_STRUCT_Pv
+    SETALIAS FramedAllocateString, _Z20FramedAllocateStringj
+    //SETALIAS g_pStringClass, g_pStringClass
+    SETALIAS JIT_NewArr1, _Z11JIT_NewArr1P21CORINFO_CLASS_STRUCT_i
+    SETALIAS CopyValueClassUnchecked, _Z23CopyValueClassUncheckedPvS_P11MethodTable
+
+    //IMPORT $JIT_New
+    //IMPORT $JIT_Box
+    //IMPORT $FramedAllocateString
+    //IMPORT $g_pStringClass
+    //IMPORT $JIT_NewArr1
+    //IMPORT $CopyValueClassUnchecked
+    //IMPORT SetAppDomainInObject
+
+
+    //IMPORT JIT_GetSharedNonGCStaticBase_Helper
+    //IMPORT JIT_GetSharedGCStaticBase_Helper
+
+
+    //EXPORT JIT_TrialAllocSFastMP_InlineGetThread__PatchTLSOffset
+    //EXPORT JIT_BoxFastMP_InlineGetThread__PatchTLSOffset
+    //EXPORT AllocateStringFastMP_InlineGetThread__PatchTLSOffset
+    //EXPORT JIT_NewArr1VC_MP_InlineGetThread__PatchTLSOffset
+    //EXPORT JIT_NewArr1OBJ_MP_InlineGetThread__PatchTLSOffset
+
+    //EXPORT JIT_GetSharedNonGCStaticBase__PatchTLSLabel
+    //EXPORT JIT_GetSharedNonGCStaticBaseNoCtor__PatchTLSLabel
+    //EXPORT JIT_GetSharedGCStaticBase__PatchTLSLabel
+    //EXPORT JIT_GetSharedGCStaticBaseNoCtor__PatchTLSLabel
+
+.macro PATCHABLE_INLINE_GETTHREAD reg, label
+\label:
+    mrc p15, 0, \reg, c13, c0, 2
+    ldr \reg, [\reg, #0xe10]
```

Comment:
This sequence is Windows-specific - 0xe10 is offset into Windows-specific thread-local OS structures.

Delete all assembly helpers that use PATCHABLE_INLINE_GETTHREAD and PATCHABLE_INLINE_GETAPPDOMAIN, and use their C++ equivalents instead. It is what we have done for bring up of Linux AMD64.